R/run.R

```diff
@@ -185,3 +316,102 @@ safe <- function(expr, ..., show = TRUE, env = safe_env()) {
     })
   })
 }
+
+can_run_rstudio_job <- function(stop_if_not = FALSE) {
+  if (!rlang::is_interactive()) {
+    return(FALSE)
+  }
+
+  has_needed_pkgs <- vapply(
+    c("rstudioapi", "httpuv"), requireNamespace, logical(1), quietly = TRUE
+  )
+
+  if (any(!has_needed_pkgs)) {
+    if (isTRUE(stop_if_not)) {
+      pkgs <- c("rstudioapi", "httpuv")[!has_needed_pkgs]
+      msg_err <- paste(
+        ngettext(
+          length(pkgs),
+          sprintf("The %s package is", pkgs[1]),
+          sprintf("The %s packages are", knitr::combine_words(pkgs))
+        ),
+        "required to run a tutorial as an RStudio job."
+      )
+      pkgs <- paste(pkgs, collapse = '", "')
+      msg <- c(msg_err, "i" = sprintf('install.packages(c("%s"))', pkgs))
+      rlang::abort(msg)
```

Comment:
`rlang` has this logic implemented and allows for safe followthrough.

```suggestion
      rlang::check_installed(c("rstudioapi", "httpuv"), "Required to run a tutorial as an RStudio job")
```